test/unit/argument_parser/detail/format_help_test.cpp

```diff
@@ -204,6 +204,65 @@ TEST(help_page_printing, do_not_print_hidden_options)
                               expected | std::views::filter(!seqan3::is_space)));
 }
 
+TEST(help_page_printing, advanced_options)
+{
+    int32_t option_value{5};
+    bool flag_value{};
+
+    auto set_up = [&option_value, &flag_value] (seqan3::argument_parser & parser)
+    {
+        parser.add_section("default section");
+        parser.add_option(option_value, 'i', "int", "this is a int option.", seqan3::option_spec::DEFAULT);
+        parser.add_section("advanced section", seqan3::option_spec::ADVANCED);
+        parser.add_subsection("advanced subsection", seqan3::option_spec::ADVANCED);
+        parser.add_option(option_value, 'j', "jnt", "this is a int option.", seqan3::option_spec::ADVANCED);
+        parser.add_flag(flag_value, 'f', "flag", "this is a flag.", seqan3::option_spec::ADVANCED);
+        parser.add_list_item("-s, --some", "list item.", seqan3::option_spec::ADVANCED);
+        parser.add_line("some line.", true, seqan3::option_spec::ADVANCED);
```

Comment:
You said something about required options not considered. Is this tested somewhere? Are the correct behaviour of hidden options tested somewhere?

Reply:
now it is :D

Reply:
And required options? How is the behaviour different here?

Reply:
required options are already tested. 
For options/flags the REQUIRED spec leads to a note in the help page. This is already tested. It also implies a specific parsing behaviour which is tested in `format_parse_test`.
For `add_section` etc, the REQUIRED flag has no effect and is ignored since it makes no sense to specify a section name as "required". So they this is the same as if DEFAULT has been used. I can add a test although it might be redundant.

Reply:
no than it is fine. Just wanted to make sure. Thanks